text/nodejs/0014-yarn-berry.md

```diff
@@ -0,0 +1,230 @@
+# Support for Yarn "Berry" 
+
+## Summary
+
+The Paketo Node.js Buildpack supports building apps which use
+[Yarn](https://yarnpkg.com/) as a package/project management tool. Over the
+last few years, the developers of Yarn have released new major versions (v2 and
+greater), collectively called Yarn "Berry", which represent a completely new
+direction for managing dependencies in Node.js projects. Chief among the
+changes is the introduction of [Plug 'n
+Play](https://yarnpkg.com/features/pnp), a system which removes the reliance on
+`node_modules` and can eliminate the need to run `yarn install` at all. The
+Yarn implementation buildpacks do not currently support building apps which use
+Berry.  
+
+There have been quite a number of changes in the behaviour of Yarn with the
+advent of Berry. Along with performance improvements and optimizations, many
+CLI operations and methods of configuration have been deprecated or removed,
+including those crucial to the functioning of the buildpack as it exists today. 
+
+* _Throughout this document, Yarn Berry (> 2.x) and Yarn Classic (1.x) will be
+  referred to as "Berry" and "Classic" respectively._
+
+## Proposal
+
+Berry seems to represent the direction in which the project is headed, as
+[stated by its lead
+maintainer](https://dev.to/arcanis/introducing-yarn-2-4eh1#conclusion). Given
+this, and the extent of the changes to the CLI, this RFC proposes that Berry
+workflows be included in the `yarn-install` buildpack under two separate codepaths/packages.
+
+## Motivation
+
+As developers continue to adopt Berry, demand for the Node.js buildpack to
+support it has steadily increased.
+
+
+## Implementation
+
+
+### API
+
+`yarn-install`
+
+Provides: `node_modules` OR `yarn_pkgs`
+Requires: `node`, `yarn` during `build`
+
+Though the Berry CLI is a complete departure from Classic, the latter may still
+be used as a global orchestrator for per-project Berry installations. The
+[recommended
+workflow](https://yarnpkg.com/getting-started/install#install-corepack) is to
+use Corepack (shipped with Node.js) to manage package manager installations,
+but this feature is experimental and arguably unnecessary in the context of
+buildpack builds where there is only one installation to manage. There are also
+challenges with supporting offline use cases using the Corepack workflow (see
+Rationale & Alternatives). For now, the buildpack should require `yarn` to
+preserve existing functionality & reduce initial engineering cost.
+
+### Detection
+
+The buildpack's detection criteria will remain unchanged, except for the
+replacement of the `node_modules` keyword with `node_pkgs`.
```

Comment:
I don't think this matches the described API below anymore.

Reply:
@ryanmoran good catch. I've updated that now.